Vagrantfile

```diff
@@ -2,31 +2,180 @@
 # vi: set ft=ruby :
 
 Vagrant.configure(2) do |config|
-  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
```

Comment:
I discussed the Vagrant approach with treydock in this [issue](https://github.com/treydock/puppet-module-keycloak/issues/132). He suggested having the domain mode Vagrant setup under _examples/vagrant-cluster_. I like that approach as well, as that way the existing Vagrant setup in this module would not be affected at all.

Reply:
Of course I have my own opinion about the directory name:  _"examples/domain-mode"_ :smile:.

Reply:
Naming, that's too hard a problem.

Reply:
Yes, Therefore you may disregard all my naming suggestions if you like :smile: .